docs/changelog/123188.yaml

```
@@ -0,0 +1,14 @@
+pr: 123188
+summary: Add deprecation warning to `TransportHandshaker`
+area: Infra/Core
+type: deprecation
+issues: []
+deprecation:
+  title: Add deprecation warning to `TransportHandshaker`
```

I still worry that end-users might not really understand this. How about something like this?

```suggestion
  title: Deprecate ability to connect to nodes of versions 8.17 and earlier
```